api/api.go

```diff
@@ -61,14 +61,23 @@ func New(mgr manager.Manager) (*api, error) {
 	return a, nil
 }
 
+func (a *api) startServer() error {
+	conf := config.Get()
+	if conf.TLSCertificate != "" && conf.TLSKey != "" {
+		return a.e.StartTLS(a.Address, conf.TLSCertificate, conf.TLSKey)
```

Comment:
Using the same port for both plain and encrypted traffic may be messy and lead to harder configuration of Deployment for this API. What about to create a new address (at another port, e.g `9993`) and use it whenever the certificate and key have been provided?